pkg/guid/guid.go

```diff
@@ -0,0 +1,41 @@
+package guid
+
+import (
+	"fmt"
+	"net"
+)
+
+// GUID address is an uint64 encapsulation for network hardware address
+type GUID uint64
+
+const guidLength = 8
+
+// ParseGUID parses string only as GUID 64 bit
+func ParseGUID(s string) (GUID, error) {
+	ha, err := net.ParseMAC(s)
+	if err != nil {
+		return 0, err
+	}
+	if len(ha) != guidLength {
+		return 0, fmt.Errorf("invalid GUID address %s", s)
+	}
+
+	return GUID(uint64(ha[0])<<(7*guidLength) | uint64(ha[1])<<(6*guidLength) | uint64(ha[2])<<(5*guidLength) | uint64(ha[3])<<(4*guidLength) |
+		uint64(ha[4])<<(3*guidLength) | uint64(ha[5])<<(2*guidLength) | uint64(ha[6])<<guidLength | uint64(ha[7])), nil
+}
+
+// String returns the string representation of GUID
+func (g GUID) String() string {
+	return g.HardWareAddress().String()
+}
+
+func (g GUID) HardWareAddress() net.HardwareAddr {
+	value := uint64(g)
+	ha := make(net.HardwareAddr, guidLength)
+	for idx := 7; idx >= 0; idx-- {
+		ha[idx] = byte(value % 0x100)
```

Comment:
please use bitwise operations

Reply:
endianness should be taken into account if this will run on big endian architecture (e.g ppc64 big endian)

Reply:
I agree change it to bitwise operation. Regarding the big endian we done in the future per arch.
Also add in the redame that we support x86 arch only